alf/experience_replayers/replay_buffer.py

```diff
@@ -41,8 +63,87 @@ def __init__(self,
             allow_multiprocess=allow_multiprocess,
             name=name)
 
+        self._prioritized_sampling = prioritized_sampling
+        if prioritized_sampling:
+            self._mini_batch_length = 1
+            tree_size = self._max_length * num_environments
+            self._sum_tree = SumSegmentTree(tree_size)
+            self._max_tree = MaxSegmentTree(tree_size)
+            self._min_tree = MinSegmentTree(tree_size)
+
     def add_batch(self, batch, env_ids=None, blocking=False):
-        self.enqueue(batch, env_ids, blocking=blocking)
+        with alf.device(self._device):
+            env_ids = self.check_convert_env_ids(env_ids)
+            self.enqueue(batch, env_ids, blocking=blocking)
+            if self._prioritized_sampling:
+                self._initialize_priority(env_ids)
+
+    @property
+    def initial_priority(self):
+        """The initial priority used for newly added experiences."""
+        return torch.max(self._max_tree.summary(), torch.tensor(1.0))
+
+    def _initialize_priority(self, env_ids):
+        last_pos = (self._current_pos[env_ids] -
+                    self._mini_batch_length) % self._max_length
+        indices = self._env_id_pos_to_index(env_ids, last_pos)
+        values = torch.full(indices.shape, self.initial_priority)
+
+        if self._mini_batch_length > 1:
+            last_pos = (self._current_pos[env_ids] - 1) % self._max_length
```

Comment:
Why this additional value updating? Seems that it sets the end of a mini_batch_length to 0 but doesn't set the middle values.